medicines/api/src/document.rs

```diff
@@ -118,17 +121,47 @@ pub async fn get_documents(
     ))
 }
 
+fn build_document_types_filter_set(document_types: Option<Vec<String>>) -> AzureFilterSet {
+    match document_types {
+        Some(document_types) => AzureFilterSet {
+            boolean_operator: "or".to_string(),
+            field_filters: document_types
+                .into_iter()
+                .map(|document_type| AzureFieldFilter {
+                    field_name: "doc_type".to_string(),
+                    operator: "eq".to_string(),
+                    field_value: format!(
+                        "{}{}",
+                        document_type[..1].to_uppercase(),
+                        document_type[1..].to_lowercase()
+                    ),
+                })
+                .collect(),
+        },
+        None => AzureFilterSet {
```

Comment:
is it possible just to return `None` here instead? also – does this even work, or does azure freak out at being told to use `$filter=or`?

Reply:
I've tested without any filters and it's working as expected.

I _could_ return `None` here, but that's going to require either:

 * `search_with_pagination_and_filter` on the search client to accept an `Option<AzureFilterSet>` which is a bit messy; or
 * an extra level of branching within `get_documents`.

I'm happy to implement either of those alternatives, but I feel like this is the neatest.